apm-agent-attach/src/main/java/co/elastic/apm/attach/JvmDiscoverer.java

```diff
@@ -96,19 +99,45 @@ private static String getJpsOutput() throws IOException, InterruptedException {
             Set<JvmInfo> set = new HashSet<>();
             for (String s : jpsOutput.split("\n")) {
                 JvmInfo parse = JvmInfo.parse(s);
-                set.add(parse);
+                // ignore jps command that we just started as it's already terminated and not relevant for attachment
+                if(!parse.packageOrPathOrJvmProperties.contains("sun.tools.jps.Jps")){
```

Comment:
[minor] might make it more resilient when packages are renamed.
```suggestion
                if(!parse.packageOrPathOrJvmProperties.contains(".Jps")){
```